test/OpenTelemetry.Instrumentation.MassTransit.Tests/MassTransitInstrumentationTests.cs

```diff
@@ -0,0 +1,143 @@
+using System.Threading.Tasks;
+using MassTransit.Testing;
+using Moq;
+using OpenTelemetry.Trace;
+using Xunit;
+
+namespace OpenTelemetry.Instrumentation.MassTransit.Tests
+{
+    public class MassTransitInstrumentationTests
+    {
+        [Fact]
+        public async Task MassTransitInstrumentationConsumerAndHandlerTest()
+        {
+            var activityProcessor = new Mock<ActivityProcessor>();
+            using (Sdk.CreateTracerProviderBuilder()
+                .AddProcessor(activityProcessor.Object)
+                .AddMassTransitInstrumentation()
+                .Build())
+            {
+                var harness = new InMemoryTestHarness();
+                var consumerHarness = harness.Consumer<TestConsumer>();
+                var handlerHarness = harness.Handler<TestMessage>();
+                await harness.Start();
+                try
+                {
+                    await harness.InputQueueSendEndpoint.Send<TestMessage>(new
+                    {
+                        Text = "Hello, world!",
+                    });
+
+                    Assert.True(await harness.Consumed.SelectAsync<TestMessage>().Any());
+                    Assert.True(await consumerHarness.Consumed.SelectAsync<TestMessage>().Any());
+                    Assert.True(await handlerHarness.Consumed.SelectAsync().Any());
+                }
+                finally
+                {
+                    await harness.Stop();
+                }
+            }
+
+            Assert.Equal(10, activityProcessor.Invocations.Count);
+
+            var sends = this.GetActivitiesFromInvocationsByOperationName(activityProcessor.Invocations, OperationName.Transport.Send);
+            var receives = this.GetActivitiesFromInvocationsByOperationName(activityProcessor.Invocations, OperationName.Transport.Receive);
+            var consumes = this.GetActivitiesFromInvocationsByOperationName(activityProcessor.Invocations, OperationName.Consumer.Consume);
+            var handles = this.GetActivitiesFromInvocationsByOperationName(activityProcessor.Invocations, OperationName.Consumer.Handle);
+
+            foreach (var activity in sends)
+            {
+                Assert.Equal("SEND /input_queue", activity.DisplayName);
+            }
+
+            foreach (var activity in receives)
+            {
+                Assert.Equal("RECV /input_queue", activity.DisplayName);
+            }
+
+            foreach (var activity in consumes)
+            {
+                Assert.Equal("CONSUME OpenTelemetry.Instrumentation.MassTransit.Tests.TestConsumer", activity.DisplayName);
+            }
+
+            foreach (var activity in handles)
+            {
+                Assert.Equal("HANDLE TestMessage/OpenTelemetry.Instrumentation.MassTransit.Tests", activity.DisplayName);
+            }
+        }
+
+        [Fact]
+        public async Task MassTransitInstrumentationTestOptions()
+        {
+            using Activity activity = new Activity("Parent");
```

Comment:
This `TracedOperations` option is troublesome. This particular test configures only the OperationName.Consumer.Consume Activity for capture. The problem is, it's the 3rd one in the Activity chain. Because it is not included in the list, the first Activity is not passed to the ActivitySourceAdapater, and is therefore not sampled. Since the root is not sampled, none of the children will be regardless of the option. Depends on the sampler, but this is the default behavior (now, it probably worked before). I added a recorded parent so the test passes, but I'm not totally convinced this feature will work that well. I kind of felt like it should be removed.

/cc @cijothomas @SergeyKanzhelev